app/models/user.rb

```diff
@@ -83,7 +83,7 @@ class User < ApplicationRecord
 
   has_one :invite_request, class_name: 'UserInviteRequest', inverse_of: :user, dependent: :destroy
   accepts_nested_attributes_for :invite_request, reject_if: ->(attributes) { attributes['text'].blank? && !Setting.require_invite_text }
-  validates :invite_request, presence: true, on: :create, if: -> { Setting.require_invite_text && !invited? }
+  validates :invite_request, presence: true, on: :create, if: -> { Setting.require_invite_text && !invited? && !external? && !bypass_invite_request_check? }
```

Comment:
I think this should not all be in a proc. Please make it a method like `invite_text_required?`